app/services/jbpm_process_service.rb

```diff
@@ -19,6 +19,25 @@ def signal(decision)
     end
   end
 
+  def valid_request?
+    ContextService.new(request.context).as_org_admin do
+      group = Group.find(request.group_ref)
+
+      if request.workflow.group_refs.empty? || group.users.empty? || !group.exists? || !group.has_role?('Approval Approver')
+        ActsAsTenant.with_tenant(Tenant.find(request.tenant_id)) do
+          ActionCreateService.new(request.id).create(
+            :operation    => Action::ERROR_OPERATION,
+            :processed_by => 'system',
+            :comments     => 'Invalid groups, either not exist or no approver role assigned'
```

Comment:
@hsong-rh can we distinguish the 2 errors separately
(1) Does the group exist in the RBAC database
(2) Does the group have the Approval Role 

And when we issue the error we need to know which is true.
There is a call in the common gem that takes in a set of groups and checks if they exist

https://github.com/RedHatInsights/insights-api-common-rails/blob/552341c359739f92d62fcb73189197b5e53283f7/lib/insights/api/common/rbac/utilities.rb#L6

We are using that in catalog, we are working with the RBAC team to have them allow us to do filtering based on an array of group uuids.
https://projects.engineering.redhat.com/browse/RHCLOUD-4307

This should make it faster.

Reply:
Separate messages for different cases.